CHANGES/1657.bugfix

```diff
@@ -0,0 +1 @@
+Disallowed anonymous users to pull new content via a pull-through caching distribution.
```

Comment:
can you add that anonymous users could still pull 'cached' content?

Reply:
I did not read through the code, but how this scenario will be handled:
Anonymos user pulls content (from existing repo) but there is an updated content remotely? We should not be allowing this, because as a result of such operation a new repo version will be created and more bits will be pulled down into pulp

Reply:
> Anonymos user pulls content (from existing repo) but there is an updated content remotely?

Good catch! I may want to introduce another check in https://github.com/pulp/pulp_container/blob/da8a70f6306b68bc5bdbd66dd0879628a30e1b43/pulp_container/app/registry_api.py#L1031 and https://github.com/pulp/pulp_container/blob/da8a70f6306b68bc5bdbd66dd0879628a30e1b43/pulp_container/app/registry.py#L132. Is there a guarantee that the same headers are passed during the redirect? With BasicAuth, we would want to be sure that the user headers remain the same.

> can you add that anonymous users could still pull 'cached' content?

Is "new content" not explicit enough?